### PR TITLE
refactor: move provider logic to status-go

### PR DESCRIPTION
**Pull request by richard-ramos**
Requires
- https://github.com/status-im/status-go/pull/2465

### CODE REVIEW

File: status/status.nim [hunk 1]
```diff
@@ -30,7 +30,6 @@ type Status* = ref object
   settings*: SettingsModel
   mailservers*: MailserversModel
   tokens*: TokensModel
-  provider*: ProviderModel
```
Comment (alaibe) [inline status/status.nim:33]: I know it is deprecated but you can also remove the file? if not, not big deal cause we are not using that in new architecture anyway

Comment (richard-ramos) [reply status/status.nim:33]: It is still used in vendor/status-lib/status/updates.nim. I'll remove it once the ens content hash decode function is implemented! (i'm still missing that for the refactoring!)

**Review by 0x-r4bbit — APPROVED** [review]
Would be cool if you could link the corresponding status-go PR in the future, so it's easier to verify the correctness of this PR

**Comment by richard-ramos** [comment]
> Would be cool if you could link the corresponding status-go PR in the future, so it's easier to verify the correctness of this PR

oops! i forgot. I just added it to the PR

File: status/status.nim (hunk 1 above)
Comment (alaibe) [reply status/status.nim:33]: That is what I am working on!

**Pull request merged**